Movie show

coreycartercodes opened the pull request:
Create 'Movie Details' page
- Refactor from controller/service into facade/poro
- Create 3 new API calls in service
- Create Facade and add organization of methods for show/index view
- Create movie PORO and add optional arguments to initialize, add formatted_runtime
- Create show spec **Need to refactor using VCR/Webmock**

closes #8

lkriffell committed:
Add implementation of vcr into movies/index_spec.rb

CCarterCodes committed:
Remove whitespace for Rubocop

CCarterCodes committed:
Add page_num +=1 to find method

CCarterCodes committed:
Add formatting for titles class

CCarterCodes committed:
Create show movie details spec

CCarterCodes committed:
Remove unnecessary MovieService call from beginning of tests

CCarterCodes committed:
Refactor index, add show action

CCarterCodes committed:
Refactor to move between facade and movie model, add get_details

CCarterCodes committed:
change movie routes

CCarterCodes committed:
Refactor to consider facades, etc..

CCarterCodes committed:
Create show details view

CCarterCodes committed:
Refactor to create facades

CCarterCodes committed:
Refactor to create movie poro

CCarterCodes committed:
Create show action

CCarterCodes committed:
Add movie_details action

CCarterCodes committed:
Add to initialize, add formatted_runtime method

CCarterCodes committed:
Implement new details to view

CCarterCodes committed:
Add get_details, get_reviews, get_cast, find_total_review_pages actions

CCarterCodes committed:
Add show spec using hard-code, need to implement vcr/webmock

lkriffell committed:
Update index spec to work with MovieFacade

lkriffell commented on .travis.yml:
```diff
@@ -8,7 +8,7 @@ script:
 deploy:
   provider: heroku
-  api_key:
+  api_key: 5c0161f1db69f834bfa0c6a5ceee7187
```
```suggestion
  api_key: 
```
This should be able to work with the ```secure:``` right below it. We don't want to upload our API key to Github. If it doesn't work I can help you fix it but this shouldn't be necessary.

lkriffell commented on app/poros/movie.rb:
```diff
@@ -0,0 +1,28 @@
+    @genres = movie_data[:genres]
+    @summary = movie_data[:overview]
+    @cast = cast_data
+    @reviews = review_data
```
I'm wondering if we could make @cast and @reviews an array of cast and review objects

lkriffell commented on app/views/movies/index.html.erb:
```diff
@@ -1,17 +1,16 @@
+      <% end %>
+    <ol>
+  <% end %>
 </center>
```
Looking Good!

lkriffell committed:
Update README

CCarterCodes committed:
Create ReadMe file

Merge branch 'readme' of github.com:lkriffell/viewing_party into readme

CCarterCodes committed:
Add README.md file
(blank line)
(blank line)
Co-authored-by: Logan Riffell <61526543+lkriffell@users.noreply.github.com>
Co-authored-by: Corey Carter <59756152+coreycartercodes@users.noreply.github.com>

coreycartercodes committed:
Merge pull request #30 from lkriffell/vcr

Implement vcr and webmock

coreycartercodes committed:
Update .travis.yml

Co-authored-by: Logan Riffell <61526543+lkriffell@users.noreply.github.com>

lkriffell committed:
Merge branch 'main' into readme

lkriffell committed:
Merge pull request #34 from lkriffell/readme

Readme

lkriffell committed:
Update README

CCarterCodes committed:
Create POROs for review/actor

CCarterCodes committed:
Refactor to include new POROs

CCarterCodes committed:
Create PORO for genre

CCarterCodes committed:
Implement genre PORO

CCarterCodes committed:
New integration of POROs

Merge branch 'movie_show' of github.com:lkriffell/viewing_party into movie_show

CCarterCodes committed:
Refactor to remove whitespace

CCarterCodes committed:
Implement VCR/Webmock

CCarterCodes committed:
Implement vcr, ignore files in coverage(spec_helper)
